Core/src/org/sleuthkit/autopsy/geolocation/GeoFilterPanel.java

```diff
@@ -415,19 +490,20 @@ int getMostRecentNumDays() {
          * 
          * @throws TskCoreException 
          */
-        private boolean isGPSDataSource(SleuthkitCase sleuthkitCase, DataSource dataSource) throws TskCoreException {
+        private Map<ARTIFACT_TYPE, Long> getGPSDataSources(SleuthkitCase sleuthkitCase, DataSource dataSource) throws TskCoreException {
+            HashMap<ARTIFACT_TYPE, Long> ret = new HashMap<>();
             for (BlackboardArtifact.ARTIFACT_TYPE type : GPS_ARTIFACT_TYPES) {
-                if (sleuthkitCase.getBlackboardArtifactsTypeCount(type.getTypeID(), dataSource.getId()) > 0) {
-                    return true;
+                long count = sleuthkitCase.getBlackboardArtifactsTypeCount(type.getTypeID(), dataSource.getId());
+                if (count > 0) {
+                    ret.put(type, count);
```

Comment:
Why are you keeping count?  You are not adding the type unless count is greater than one.  I just don't think you need to keep count.

Reply:
We display the count for each artifact type in the UI, so that's why I'm keeping count here.